models/yasgbt.py

```diff
@@ -0,0 +1,90 @@
+import numpy as np
+from tqdm.notebook import tqdm
+
+from sklearn.base import BaseEstimator
+from sklearn.ensemble import HistGradientBoostingRegressor
+
+from sksurv.functions import StepFunction
+from sksurv.nonparametric import kaplan_meier_estimator, CensoringDistributionEstimator
+
+from models.survival_mixin import SurvivalMixin
+
+
+class DataSampler:
```

Comment:
Let's call this `IBSTrainingSampler` to make it more explicit that this sampler implements the weighting scheme of the IBS formula.